src/PrMerger.js

```diff
@@ -41,11 +53,32 @@ class PrMerger {
                     continue;
                 }
 
+                const clearedForMerge = rawPr.labels.some(el => el.name === Config.clearedForMergeLabel());
+
+                // There are usually few 'cleared' PRs (i.e., ready for merge), so do not ignore them.
+                // This should help handle sutiations where some of such 'cleared' PRs were updated just after
+                // this scan has been started. Other (non-cleared) PRs (that are not going to be merged now anyway)
+                // can be ignored until the next scan.
+                // TODO: also handle a situation when the PR becomes 'cleared' just after this scan began.
+                if (!clearedForMerge && oldScanResult && oldScanResult.isStillUnchanged(rawPr)) {
+                    Logger.info(`Ignoring PR${rawPr.number} because it has not changed since the previous scan of ${oldScanResult.scanDate.toISOString()}`);
+                    this._ignoredUnchanged++;
+                    scanResult.unchangedPrs.push(rawPr);
+                    continue;
+                }
+
                 const result = await MergeContext.Process(rawPr, somePrWasStaged);
                 assert(!somePrWasStaged || !result.prStaged());
                 somePrWasStaged = somePrWasStaged || result.prStaged();
-                if (result.delayed() && (minDelay === null || minDelay > result.delayMs()))
-                    minDelay = result.delayMs();
+                if (result.delayed() && (scanResult.delay === null || scanResult.delay > result.delayMs()))
+                    scanResult.delay = result.delayMs();
```

Comment:
I do not understand why we need to remember minDelay (beyond this function call lifetime). I would expect this code to stay intact and scanResult to have no delay field.